lib/checkPullRequestReview.js

```diff
@@ -0,0 +1,306 @@
+  //  with the pull request number and in the appropriate repository. An
+  // empty array will be returned if the pull request is not approved by
+  // all reviewers.
+  const repoSearchString = `repo:${context.payload.repository.full_name}`;
+  const reviewApprovedSearchString = 'review:approved';
+  const searchResult = await context.github.search.issuesAndPullRequests(
+    context.repo({
+      q:
+        repoSearchString +
+        ' ' +
+        reviewApprovedSearchString +
+        ' ' +
+        pullRequest.number,
+    })
+  );
+
+  const searchData = searchResult.data.items;
+  const assignees = pullRequest.assignees.map((assignee) => assignee.login);
+  if (searchData.length === 0) {
+    // Pull request has not been approved by all reviewers, hence we need to
+    // assign the remaining reviewers.
+    const reviewersYetToReview = pullRequest.requested_reviewers.map(
+      (reviewer) => reviewer.login
+    );
+
+    const unassignedReviewers = reviewersYetToReview.filter(
+      (reviewer) => !assignees.includes(reviewer)
+    );
+    if (unassignedReviewers.length > 0) {
+      // Ping and assign unassigned reviewers.
+      const commentBody =
+        'Assigning @' +
+        unassignedReviewers.join(', @') +
+        ' for code owner reviews, Thanks!';
+
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body: commentBody,
+        })
+      );
+
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: unassignedReviewers,
+        })
+      );
+    }
+  } else {
+    // Since the pull request has been approved, we add the LGTM label.
+    const labels = pullRequest.labels.map((label) => label.name);
+    if (labels.includes(LGTM_LABEL)) {
+      // Do nothing and end execution if label has already been added.
+      return;
+    }
+    await context.github.issues.addLabels(
+      context.repo({
+        issue_number: pullRequest.number,
+        labels: [LGTM_LABEL],
+      })
+    );
+
+    // The last reviewer got unassigned, but the pull request assignees list
+    // does not reflect that since we did not refetch the pull request data,
+    // hence if any other person is assigned to the pull request, they are
+    // likely the one to merge it.
+
+    if (assignees.some((user) => user !== reviewer)) {
+      const remainingAssignee = assignees.find(user => user !== reviewer);
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            remainingAssignee +
+            ', this PR is ready to be merged. We are pinging you since ' +
+            'you are the remaining assingee on this PR. Please make sure ' +
+            "there are no pending comments from the author's end before " +
+            'merge. Thanks!',
+        })
+      );
+      return;
```

Comment:
This is a bit confusing - can we be sure that the assignee has merging rights?

Why not do this instead - get list of assignees and check if any one of them has merging rights.

If yes - leave a comment asking them to do the merge.
If no - assign the reviewer.

This is simpler and more readable as well.

Reply:
Following discussions offline, we resolved to only assign the project owner if the author does not have merging rights since we are sure that all pull requests will have a changelog label.